### DISCUSSION TITLE
Idle function do not take care of Synchronizing the  multi core states during S2I and S2R.

### DISCUSSION THREAD

**Discussion by SumitJain3083** [zephyrproject-rtos/zephyr#59285] [Ideas]
In Multicore environment , if the policy manager decides that the system has to go to S2I or S2R then the idle framework will set S2I / S2R state for all the cores and once the pm_state_set is called on respective cores all the cores goes to suspend state.

or Is there a way to differentiate the last core and the other cores , so that the last core can be in different state then the other cores ?

Expected behavior
Expectation is last core shouldn't go to suspend mode in S2I / S2R.

**Comment by ceolin** [comment]
The policy made its decision per core based on the time of the next scheduled event. Much likely the policy will select the same state if these cores get idle closely. Note that states are per CPU, if one core is "special" you can have different states for it.

Other possibility is to write your own policy that checks for other active cores before deciding which state to use.
